cpp_format_tools/Dockerfile

```diff
@@ -1,16 +1,16 @@
 ARG BUILDIFIER_VERSION=3.5.0
-ARG CLANG_VERSION=10.0.1-r0
+ARG CLANG_VERSION=17.0.5-r0
```

Comment:
I am not fully familiar with the purpose of this docker image. But if it is meant to format otel-cpp, as of now we need clang-format v10 for formatting. Will this upgrade bump up the clang format too?

Reply:
oh same question by @marcalff  :)